be/src/vec/olap/olap_data_convertor.h

```diff
@@ -0,0 +1,256 @@
+        OlapFieldData get_data_at(size_t offset) const override;
+        Status convert_to_olap() override;
+
+    private:
+        bool m_check_length;
+        PaddedPODArray<Slice> m_slice;
+    };
+
+    class OlapColumnDataConvertorDate : public OlapColumnDataConvertorBase {
+    public:
+        OlapColumnDataConvertorDate() = default;
+        ~OlapColumnDataConvertorDate() override = default;
+
+        void set_source_column(const ColumnWithTypeAndName& typed_column, size_t row_pos,
+                               size_t num_rows) override;
+        const void* get_data() const override;
+        OlapFieldData get_data_at(size_t offset) const override;
+        Status convert_to_olap() override;
+
+    private:
+        PaddedPODArray<uint24_t> m_values;
+    };
+
+    class OlapColumnDataConvertorDateTime : public OlapColumnDataConvertorBase {
+    public:
+        OlapColumnDataConvertorDateTime() = default;
+        ~OlapColumnDataConvertorDateTime() override = default;
+
+        void set_source_column(const ColumnWithTypeAndName& typed_column, size_t row_pos,
+                               size_t num_rows) override;
+        const void* get_data() const override;
+        OlapFieldData get_data_at(size_t offset) const override;
+        Status convert_to_olap() override;
+
+    private:
+        PaddedPODArray<uint64_t> m_values;
+    };
+
+    class OlapColumnDataConvertorDecimal : public OlapColumnDataConvertorBase {
+    public:
+        OlapColumnDataConvertorDecimal() = default;
+        ~OlapColumnDataConvertorDecimal() override = default;
+
+        void set_source_column(const ColumnWithTypeAndName& typed_column, size_t row_pos,
+                               size_t num_rows) override;
+        const void* get_data() const override;
+        OlapFieldData get_data_at(size_t offset) const override;
+        Status convert_to_olap() override;
+
+    private:
+        PaddedPODArray<decimal12_t> m_values;
+    };
+
+    // class OlapColumnDataConvertorSimple for simple types, which don't need to do any convert, like int, float, double, etc...
+    template <typename T>
+    class OlapColumnDataConvertorSimple : public OlapColumnDataConvertorBase {
+    public:
+        OlapColumnDataConvertorSimple() = default;
+        ~OlapColumnDataConvertorSimple() override = default;
+
+        void set_source_column(const ColumnWithTypeAndName& typed_column, size_t row_pos,
+                               size_t num_rows) override {
+            OlapBlockDataConvertor::OlapColumnDataConvertorBase::set_source_column(
+                    typed_column, row_pos, num_rows);
+            m_values.resize(num_rows);
+        }
+
+        const void* get_data() const override { return m_values.data(); }
+
+        OlapFieldData get_data_at(size_t offset) const override {
+            assert(offset < m_num_rows && m_num_rows == m_values.size());
+            UInt8 null_flag = 0;
+            auto null_map = get_nullmap();
```

Comment:
each time have a virtual function call ? why not keep the `_null_map` ptr as Member variable?